tax_from_account/models/tax_from_account.py

```diff
@@ -5,84 +5,82 @@
 
 
 def _get_default_taxes(line, partner=None, inv_type="out_invoice"):
-    InvoiceLine = line.env["account.invoice.line"]
     company = line.company_id or line.order_id.company_id
     fpos = (
         line.order_id.fiscal_position_id
         or line.order_id.partner_id.property_account_position_id
     )
-    account = InvoiceLine.get_invoice_line_account(
-        inv_type, line.product_id, fpos, company
-    )
-
-    tax_field = "taxes_id" if inv_type.startswith("out_") else "supplier_taxes_id"
-
     company_tax_field = (
         "account_sale_tax_id"
         if inv_type.startswith("out_")
         else "account_purchase_tax_id"
     )
-    # Don't try to collapse the filtering, needs independent evaluation of
-    # each possibility.
-    taxes = (
-        line.product_id[tax_field].filtered(lambda t: t.company_id == company)
-        or account.tax_ids.filtered(lambda t: t.company_id == company)
-        or company[company_tax_field]
-    )
-    return fpos.map_tax(taxes, line.product_id, partner) if fpos else taxes
+    tax_field = "taxes_id" if inv_type.startswith("out_") else "supplier_taxes_id"
+    account_type = "income" if inv_type.startswith("out_") else "expense"
+    tax_ids = line.env["account.tax"]
+    if line.product_id[tax_field].filtered(lambda tax: tax.company_id == company):
+        tax_ids = line.product_id[tax_field].filtered(
+            lambda tax: tax.company_id == company
+        )
+    elif line.product_id.product_tmpl_id.get_product_accounts(fiscal_pos=fpos)[
```

Comment:
No, can't do it like this. In fact the original code is more sensible. Just get the account and reuse the statement to get taxes you deleted.